apps/studio/src/pages/sites/[siteId]/index.tsx

```diff
@@ -22,12 +26,22 @@ import { CreatePageModal } from "~/features/editing-experience/components/Create
 import { MoveResourceModal } from "~/features/editing-experience/components/MoveResourceModal"
 import { useQueryParse } from "~/hooks/useQueryParse"
 import { type NextPageWithLayout } from "~/lib/types"
+import { buildPermissionsFor } from "~/server/modules/permissions/permissions.util"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
+import { trpc } from "~/utils/trpc"
 
 export const sitePageSchema = z.object({
   siteId: z.coerce.number(),
 })
 
+const getPermissions = (roles: { role: RoleType }[]) => {
```

Comment:
why are we doing this logic on FE ah?

Reply:
this is just the builder! we need it because i tried sending the `builder.rules` across but it didn't work. i'll try the `pack/unpack` method in a follow-up